front/pages/api/v1/w/[wId]/assistant/agent_configurations/[sId]/index.ts

```diff
@@ -0,0 +1,192 @@
+import type { GetOrPatchAgentConfigurationResponseType } from "@dust-tt/client";
+import { PatchAgentConfigurationRequestSchema } from "@dust-tt/client";
+import type { WithAPIErrorResponse } from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { getLightAgentConfiguration } from "@app/lib/api/assistant/configuration";
+import { setAgentUserFavorite } from "@app/lib/api/assistant/user_relation";
+import { withPublicAPIAuthentication } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { apiError } from "@app/logger/withlogging";
+/**
+ * @swagger
+ * /api/v1/w/{wId}/assistant/agent_configurations/{sId}:
+ *   get:
+ *     summary: Get agent configuration
+ *     description: Retrieve the agent configuration identified by {sId} in the workspace identified by {wId}.
+ *     tags:
+ *       - Assistants
+ *     parameters:
+ *       - in: path
+ *         name: wId
+ *         required: true
+ *         description: ID of the workspace
+ *         schema:
+ *           type: string
+ *       - in: path
+ *         name: sId
+ *         required: true
+ *         description: ID of the agent configuration
+ *         schema:
+ *           type: string
+ *     security:
+ *       - BearerAuth: []
+ *     responses:
+ *       200:
+ *         description: Successfully retrieved agent configuration
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 agentConfiguration:
+ *                   $ref: '#/components/schemas/AgentConfiguration'
+ *       400:
+ *         description: Bad Request. Invalid or missing parameters.
+ *       401:
+ *         description: Unauthorized. Invalid or missing authentication token.
+ *       404:
+ *         description: Agent configuration not found.
+ *       405:
+ *         description: Method not supported. Only GET or PATCH is expected.
+ *       500:
+ *         description: Internal Server Error.
+ *   patch:
+ *     summary: Update agent configuration
+ *     description: Update the agent configuration identified by {sId} in the workspace identified by {wId}.
+ *     tags:
+ *       - Assistants
+ *     parameters:
+ *       - in: path
+ *         name: wId
+ *         required: true
+ *         description: ID of the workspace
+ *         schema:
+ *           type: string
+ *       - in: path
+ *         name: sId
+ *         required: true
+ *         description: ID of the agent configuration
+ *         schema:
+ *           type: string
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *            "requestBody": {
+ *               "type": "object",
+ *               "properties": {
+ *                 "userFavorite": {
```

Comment:
Should we start using snake_case for new endpoints?

Reply:
I overheard the discussions around this.
I don't have a strong opinion on the casing style but I don't think we should sneak in a new pattern & I think it's out of scope.

To me, if we decide to change the style of the public api args, we should do it all at once (I believe we can do some magic via zod to handle the old style transparently).

Do you want to open a "Decision" around this ?

Reply:
Agreed consistency is likely better now that we knee deep into it.

No extreme urgency on doing it, we'll likely tackle it soon in a broader project aiming at giving some love to our API and our integrations with platforms like zapier and like

Reply:
> we'll likely tackle it soon in a broader project aiming at giving some love to our API and our integrations with platforms like zapier and like

100%, I'm quite interested in improving it.

TBH, we are already giving a lot of love to the API (after breaking it a bit) :)